src/sentry/integrations/pagerduty/integration.py

```diff
@@ -0,0 +1,169 @@
+from __future__ import absolute_import
+
+from django.utils.translation import ugettext_lazy as _
+
+from sentry import options
+
+from sentry.utils import json
+from sentry.integrations import (
+    IntegrationInstallation,
+    IntegrationFeatures,
+    IntegrationMetadata,
+    IntegrationProvider,
+    FeatureDescription,
+)
+from sentry.models import PagerDutyServiceProject, Project
+from sentry.pipeline import PipelineView
+
+DESCRIPTION = """
+PagerDuty Description
+"""
+
+FEATURES = [
+    FeatureDescription(
+        """
+        Configure rule based PagerDuty notifications!!
+        """,
+        IntegrationFeatures.ALERT_RULE,
+    )
+]
+
+metadata = IntegrationMetadata(
+    description=_(DESCRIPTION.strip()),
+    features=FEATURES,
+    author="The Sentry Team",
+    noun=_("Installation"),
+    issue_url="https://github.com/getsentry/sentry/issues/new?title=Slack%20Integration:%20&labels=Component%3A%20Integrations",
+    source_url="https://github.com/getsentry/sentry/tree/master/src/sentry/integrations/slack",
+    aspects={},
+)
+
+
+class PagerDutyIntegration(IntegrationInstallation):
+    def get_client(self):
+        pass
+
+    def get_organization_config(self):
+        from sentry.models import Project
+
+        projects = Project.objects.filter(
+            organization_id__in=self.model.organizations.values_list("id", flat=True)
+        )
+        items = []
+        for p in projects:
+            items.append({"value": p.id, "label": p.name})
+
+        service_options = [(s['id'], s['name'],) for s in self.services.values()]
+
+        fields = [
+            {
+                "name": "project_mapping",
+                "type": "choice_mapper",
+                "label": _("Map projects in Sentry to services in PagerDuty"),
+                "help": _(
+                    "When an alert rule is triggered in a project, this mapping will let us know what service to create the incident in PagerDuty."
+                ),
+                "addButtonText": _("Add Sentry Project"),
+                "addDropdown": {
+                    "emptyMessage": _("All projects configured"),
+                    "noResultsMessage": _("Could not find project"),
+                    "items": items,
+                },
+                "mappedSelectors": {
+                    "service": {
+                        "choices": service_options,
+                        "placeholder": _("Select a service"),
+                    }
+                },
+                "columnLabels": {"service": _("Service")},
+                "mappedColumnLabel": _("Sentry Project"),
+            }
+        ]
+
+        return fields
+
+    def update_organization_config(self, data):
+
+        if "project_mapping" in data:
+            project_ids_and_services = data.pop("project_mapping")
+
+            PagerDutyServiceProject.objects.filter(
+                organization_integration_id=self.org_integration.id,
+            ).delete()
```

Comment:
Should this and the insert be in a transaction.atomic() block so we can ensure that they either both happen or neither does?